util/cmake/SeqAnBuildSystem.cmake

```diff
@@ -180,41 +173,78 @@ macro (seqan_build_system_init)
+    endif ()
+    if (COMPILER_IS_INTEL)
+        # disable some warnings on ICC
+        set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -wd3373,2102")
+    endif (COMPILER_IS_INTEL)
+    if (MSVC)
+        # Use the /W2 warning level for visual studio.
+        set (CMAKE_CXX_WARNING_LEVEL 2) # TODO(h-2): raise this to W4
+        # Disable warnings about unsecure (although standard) functions.
+        set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} /D_SCL_SECURE_NO_WARNINGS")
+    endif ()
+
+    # DEFFAULT BUILD SYSTEM
     if (NOT SEQAN_BUILD_SYSTEM)
-        set (SEQAN_BUILD_SYSTEM "DEVELOP")
+        set (SEQAN_BUILD_SYSTEM "DEVELOP" CACHE STRING "Build/Release mode to select. One of DEVELOP SEQAN_RELEASE, APP:\${APP_NAME}. Defaults to DEVELOP.")
     endif (NOT SEQAN_BUILD_SYSTEM)
     set (SEQAN_APP_VERSION "0.0.0" CACHE STRING "Version of the application.")
     set (SEQAN_NIGHTLY_RELEASE FALSE CACHE BOOL "Set to TRUE to enable nightly app releases.")
 
-    if (("${SEQAN_BUILD_SYSTEM}" STREQUAL "SEQAN_RELEASE") OR
-        ("${SEQAN_BUILD_SYSTEM}" STREQUAL "SEQAN_RELEASE_LIBRARY"))
-        # Install SeqAn LICENSE, README.rst, CHANGELOG.rst files.
-        install (FILES LICENSE
-                       README.rst
-                       CHANGELOG.rst
-                 DESTINATION share/doc/seqan)
-        # Install pkg-config file, except on Windows.
-        if (NOT CMAKE_SYSTEM_NAME MATCHES Windows)
-            configure_file("util/pkgconfig/seqan.pc.in" "${CMAKE_BINARY_DIR}/util/pkgconfig/seqan-${SEQAN_VERSION_MAJOR}.pc" @ONLY)
-            install(FILES "${CMAKE_BINARY_DIR}/util/pkgconfig/seqan-${SEQAN_VERSION_MAJOR}.pc" DESTINATION share/pkgconfig)
-            # Install FindSeqAn TODO(h-2) rename FindSeqAn.cmake to FindSeqAn${SEQAN_VERSION_MAJOR}.cmake after 2.x cycle
-            install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/util/cmake/FindSeqAn.cmake" DESTINATION share/cmake/Modules)
-        endif (NOT CMAKE_SYSTEM_NAME MATCHES Windows)
+    # STANDARD BUILD FLAGS
+
+    # OpenBSD just can't handle it
+    if (${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
+        set (SEQAN_NO_NATIVE TRUE)
+        set (SEQAN_OFFICIAL_PKGS FALSE)
     endif ()
 
-    if (("${SEQAN_BUILD_SYSTEM}" MATCHES "SEQAN_RELEASE") OR
-        ("${SEQAN_BUILD_SYSTEM}" MATCHES "APP"))
+    # packages for distribution
+    if ((SEQAN_OFFICIAL_PKGS) AND
+        (NOT MSVC) AND
+        (SEQAN_BUILD_SYSTEM MATCHES "APP")) # either APP:$app or SEQAN_RELEASE_APPS
+
+        # static linkage
         set (SEQAN_STATIC_APPS TRUE CACHE INTERNAL "Create static app binaries")
+        message (STATUS "Building static binaries.")
+
+        # machine specific optimizations
+        if ((CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64") OR (CMAKE_SYSTEM_PROCESSOR STREQUAL "amd64"))
+            set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mmmx -msse -msse2 -msse3 -mssse3 -mpopcnt")
+            message (STATUS "Release binaries built with optimizations for SSE3 and POPCNT.")
```

Comment:
barcelona has SSE4a which is a different instruction set than SSE4.1/SSE4.2 on Intel. GCC also has a `-msse4` but documentation implies this might be just ==`-msse4.1 -msse4.2` so it is also not compatible.